### ISSUE TITLE
Relationship data for include missing

### ISSUE THREAD

**Issue by blmundie** [fotinakis/jsonapi-serializers#74]
I think this might be related to #47.  When including `posts` the relationship data for posts is included, but for `posts.users` the relationship data for posts is not included.  `posts,posts.users` works

The line `direct_children_includes = includes.reject { |key| key.include?('.') }` in serializer.rb:267 seems to cause this.  Changing it to `direct_children_includes = includes.reject { |key| key.split('.')[0] }` fixes this.  I couldn't tell if this is intended or not, because there is a test that checks for it.

**Comment by fotinakis** [comment]
Hmm, that's _not_ intentional. In a pre-1.0 version of the spec, the spec said to only include leaf nodes for includes, but in the 1.0 spec they changed it to include all parent nodes as well. There was a PR that fixed it (https://github.com/fotinakis/jsonapi-serializers/pull/50) and I thought we have good test coverage for making sure that this work. You're definitely using the latest version of this gem? Also, #47 that you referenced was actually caused by him overriding `format_name` and a bad interaction with strings/symbols—have you overridden format_name?

**Comment by blmundie** [comment]
Yes, I'm definitely using the latest version.  It looks to me that the tests on line 705 and 739 test for this.
